pkg/jobs/jobs.go

```diff
@@ -201,39 +197,65 @@ func (j *Job) startWorkers(ctx context.Context, workersCount int, quitCh chan st
 	var (
 		errcList = make([]<-chan *WorkerError, 0)
 		wg       sync.WaitGroup
+		// rwLock   sync.RWMutex
 	)
+	// appendToErrChClist := func(errcList []<-chan *WorkerError, errCh chan *WorkerError) {
+	// 	defer rwLock.Unlock()
+	// 	rwLock.Lock()
+	// 	errcList = append(errcList, errCh)
+	// }
+
 	wg.Add(workersCount)
 	for i := 0; i < workersCount; i++ {
 		go func(ctx context.Context, workerId int, wq WorkQueue, quitCh chan struct{}) {
 			errCh := make(chan *WorkerError, 1)
-			errcList = append(errcList, errCh)
-			defer func() {
+			defer func(errCh chan *WorkerError) {
 				quitCh <- struct{}{}
 				close(errCh)
-				klog.V(6).Infof("%s worker %d stopped\n", j.ID, workerId)
-			}()
-			klog.V(6).Infof("%s worker %d started\n", j.ID, workerId)
+				klog.V(1).Infof("%s worker %d stopped\n", j.ID, workerId)
+			}(errCh)
+			errcList = append(errcList, errCh)
+			// appendToErrChClist(errcList, errCh)
+
 			wg.Done()
+			klog.V(1).Infof("%s worker %d started\n", j.ID, workerId)
 			for {
 				var task interface{}
+				if !j.GetIsWorkerExitsOnEmptyQueue() && wq.Count() == 0 {
```

Comment:
Moving on top does not change a thing. 
However, looking at the blocking `Get` and how it should be affected by the change in the Job's `IsWorkerExitsOnEmptyQueue` state, consider the following:
Potentially, there is a situation when the worker of a job with `IsWorkerExitsOnEmptyQueue=false` is blocked on `Get` and the queue is empty. Later, you may change `IsWorkerExitsOnEmptyQueue=true`, expecting the job to exit because the queue is empty, however unless something stops the queue, there's not mechanism to unblock the Get.
That could be addressed e.g. by making `SetIsWorkerExitsOnEmptyQueue` implement or start a logic that will unblock waiting Gets.
The whole controller Stop/Shutdown logic deserves a second and more careful review and perhaps refactoring. But all that should move to its own PR. Just mentioning it. Don't do it here.